web/modules/custom/joinup_subscription/src/Form/SubscriptionDashboardForm.php

```diff
@@ -0,0 +1,175 @@
+use Drupal\joinup_subscription\JoinupSubscriptionInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a Joinup subscription form.
+ */
+class SubscriptionDashboardForm extends FormBase {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity type bundle info service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   */
+  protected $entityTypeBundleInfo;
+
+  /**
+   * The Joinup relation manager service.
+   *
+   * @var \Drupal\joinup_core\JoinupRelationManagerInterface
+   */
+  protected $relationManager;
+
+  /**
+   * Constructs a new SubscriptionDashboardForm.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entityTypeBundleInfo
+   *   The entity type bundle info service.
+   * @param \Drupal\joinup_core\JoinupRelationManagerInterface $relationManager
+   *   The Joinup relation manager service.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, EntityTypeBundleInfoInterface $entityTypeBundleInfo, JoinupRelationManagerInterface $relationManager) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityTypeBundleInfo = $entityTypeBundleInfo;
+    $this->relationManager = $relationManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('joinup_core.relations_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'joinup_subscription_dashboard';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, AccountInterface $user = NULL) {
+    // When no user is passed we cannot show anything useful.
+    if (empty($user)) {
+      throw new \InvalidArgumentException('No user account supplied.');
+    }
+
+    $memberships = $this->relationManager->getUserGroupMembershipsByBundle($user, 'rdf_entity', 'collection');
+    $bundle_info = $this->entityTypeBundleInfo->getBundleInfo('node');
+
+    $form['description'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'p',
+      '#value' => $this->t('Set your preferences to receive notifications on a per collection basis.'),
+    ];
+
+    $empty_message = $this->t('No collection memberships yet. Join one or more collections to subscribe to their content!');
+    $form['empty_text'] = [
+      '#theme' => 'status_messages',
+      '#message_list' => ['status' => [$empty_message]],
+      '#status_headings' => [
+        'status' => t('Status message'),
+        'error' => t('Error message'),
+        'warning' => t('Warning message'),
```

Comment:
s/t()/$this->t() for all #status_headings items.

Reply:
:+1: